### PR TITLE
Basic changes to integrate the new platform model based on the decomposed 'universe' BOM

### PR DESCRIPTION
**Pull request by aloubyansky**
Most of this code isn't going to be actually used until we introduce the new platform model in 2.0.
Some parts of these changes are going to change in perspective but they already allow creating projects that use extensions from multiple platform BOMs.
The PR also includes a BOM import alignment checks, i.e. whether the imported platform BOMs belong to the same platform release. In this PR, the detected misalignment results in an error. This should be controlled by a config option though.

### CODE REVIEW

**Comment by famod** [comment]
I cancelled and restarted CI after: https://www.githubstatus.com/incidents/zbpwygxwb3gw

File: devtools/gradle/src/main/java/io/quarkus/devtools/project/buildfile/GradleGroovyProjectBuildFile.java [hunk 1]
```diff
@@ -25,6 +25,11 @@ String getBuildGradlePath() {
         return BUILD_GRADLE_PATH;
     }
 
+    @Override
+    protected boolean importBom(ArtifactCoords coords) {
```
Comment (ia3andy) [inline devtools/gradle/src/main/java/io/quarkus/devtools/project/buildfile/GradleGroovyProjectBuildFile.java:29]: why not `addBom`?

File: independent-projects/bootstrap/app-model/src/test/java/io/quarkus/bootstrap/model/PlatformImportsTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        assertEquals(1, platform.getStreams().size());
+        assertEquals(1, pi.getPlatforms().size());
+
+        assertTrue(pi.isAligned(Collections.singletonMap("io.playground",
+                Arrays.asList(AppArtifactCoords.fromString("io.playground:playground-bom::pom:1.1.1"),
+                        AppArtifactCoords.fromString("io.playground:acme-bom::pom:2.2.2")))));
+        assertFalse(pi.isAligned(Collections.singletonMap("io.playground",
+                Arrays.asList(AppArtifactCoords.fromString("io.playground:playground-bom::pom:1.1.2"),
+                        AppArtifactCoords.fromString("io.playground:acme-bom::pom:2.2.2")))));
+    }
```
Comment (ia3andy) [inline independent-projects/bootstrap/app-model/src/test/java/io/quarkus/bootstrap/model/PlatformImportsTest.java:67]: You should start using AssertJ it's so much nicer :) (and also fails logs are nicer)

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute [hunk 3]
```diff
@@ -23,6 +23,9 @@ val quarkusPlatformVersion: String by project
 
 dependencies {
     implementation(enforcedPlatform("$\{quarkusPlatformGroupId}:$\{quarkusPlatformArtifactId}:$\{quarkusPlatformVersion}"))
```
Comment (ia3andy) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/base/build-layout.include.qute:25]: so we keep the legacy platform bom for now?

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/CodestartSpec.java [hunk 4]
```diff
@@ -85,20 +85,23 @@ public boolean isPreselected() {
...
         public LanguageSpec() {
-            this(null, null, null, null);
+            this(null, null, null, null, null);
         }
 
         @JsonCreator
         public LanguageSpec(@JsonProperty("data") Map<String, Object> data,
                 @JsonProperty("shared-data") Map<String, Object> sharedData,
                 @JsonProperty("dependencies") List<CodestartDep> dependencies,
+                @JsonProperty("boms") List<CodestartDep> boms,
```
Comment (ia3andy) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/CodestartSpec.java:99]: Do you see a case were a bom could be declared by a codestart? if not I would drop this, the using the inputData should be enough..

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java [hunk 5]
```diff
@@ -82,15 +78,61 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
...
+                }
+            }
+        } else {
+            bom = extensionCatalog.getBom();
+        }
+        if (bom == null) {
+            throw new QuarkusCommandException("The platform BOM is missing");
+        }
+
+        invocation.setValue(BOM_GROUP_ID, bom.getGroupId());
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:121]: could we move that logic somewhere else?

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/union/Playground.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class Playground {
+
+    public static void main(String[] args) throws Exception {
```
Comment (ia3andy) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/union/Playground.java:19]: Why not a test instead?

**Review by ia3andy — CHANGES_REQUESTED** [review]
Added a few comments ;)

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java (hunk 5 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:121]: and test it

**Comment by aloubyansky** [comment]
@glefloch fyi, I am moving the gradle model interfaces and impl from `gradle-resolver` to `app-model`. Apparently, it looks like, Gradle fails to locate model API if it's not in the same jar. It's weird. But we could perhaps benefit from this move and unify some of model API between Gradle and Maven.

**Review by ia3andy — APPROVED** [review]
Cool :)

**Pull request merged**